bundles/org.openhab.binding.smaenergymeter/src/main/java/org/openhab/binding/smaenergymeter/internal/handler/SMAEnergyMeterHandler.java

```diff
@@ -61,68 +74,121 @@ public void initialize() {
 
         EnergyMeterConfig config = getConfigAs(EnergyMeterConfig.class);
 
-        int port = (config.getPort() == null) ? EnergyMeter.DEFAULT_MCAST_PORT : config.getPort();
-        energyMeter = new EnergyMeter(config.getMcastGroup(), port);
+        int port = (config.getPort() == null) ? PacketListener.DEFAULT_MCAST_PORT : config.getPort();
+
         try {
-            energyMeter.update();
+            PacketListener thingListener = listenerRegistry.getListener(config.getMcastGroup(), port);
+            serialNumber = config.getSerialNumber();
+            if (serialNumber == null) {
+                if (!thing.getProperties().containsKey(Thing.PROPERTY_SERIAL_NUMBER)
+                        || (serialNumber = retrieveSerialNumber(thing.getProperties())) == null) {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_PENDING,
+                            "Meter serial number missing");
+                    return;
+                }
+
+                // copy serial number from thing properties into config
+                Map<String, Object> newConfig = editConfiguration().getProperties();
+                newConfig.put("serialNumber", serialNumber);
+                Thing thing = editThing().withConfiguration(new Configuration(newConfig)).build();
+                updateThing(thing);
+                updateStatus(ThingStatus.OFFLINE);
+                return;
+            } else {
+                if (!thing.getProperties().containsKey(Thing.PROPERTY_SERIAL_NUMBER)) {
+                    Map<String, String> props = editProperties();
+                    props.put(Thing.PROPERTY_VENDOR, "SMA");
+                    props.put(Thing.PROPERTY_SERIAL_NUMBER, serialNumber);
```

Comment:
You can't have at the same time a configuration parameter and a property with the same name.
As you add in that PR a configuration parameter, do not set anymore a property.